src/core/frames/form_interceptor.ts

```diff
@@ -21,8 +21,8 @@ export class FormInterceptor {
   }
 
   submitBubbled = <EventListener>((event: SubmitEvent) => {
-    if (event.target instanceof HTMLFormElement) {
-      const form = event.target
+    const form = event.target
+    if (form instanceof HTMLFormElement && form.closest("turbo-frame, html") == this.element) {
```

Comment:
What is the significance of the second `html` selector?

Also, if the frame that the `<form>` is nested within matches `turbo-frame[disabled]`, wouldn't we want events to bubble past it to the outer frame?

Reply:
According to the commit message, the `element.closest("turbo-frame, html") == this.element` is lifted directly from the `LinkInterceptor`. The `[disabled]` checks happen within the delegates' `shouldInterceptLinkClick` and `shouldInterceptFormSubmission`, which on further consideration feels more aligned with the delegator pattern than leaking that concept up into the observer layer.

Reply:
> What is the significance of the second html selector?

I had the same question about the logic in [link_interceptor.ts#L58](https://github.com/hotwired/turbo/blob/c54b443d26b525644b60ef3a04b47dba0c72ef6f/src/core/frames/link_interceptor.ts#L58), but figured I'd follow the same pattern to be safe.

> Also, if the frame that the <form> is nested within matches turbo-frame[disabled], wouldn't we want events to bubble past it to the outer frame?

I also wondered this, and do not know the answer.  Under what circumstances are a frame disabled?

I saw you were the last one that worked with this code, which is why I pinged you for review.

> The [disabled] checks happen within the delegates' shouldInterceptLinkClick and shouldInterceptFormSubmission, which on further consideration feels more aligned with the delegator pattern than leaking that concept up into the observer layer.

I originally placed the code in `shouldInterceptFormSubmission`, but then went this direction after seeing how LinkInterceptor handled it.  

```diff
private shouldInterceptNavigation(element: Element, submitter?: Element) {
  const id = submitter?.getAttribute("data-turbo-frame") || element.getAttribute("data-turbo-frame") || this.element.getAttribute("target")

- if (!this.enabled || id == "_top") {
+ if (!this.enabled || id == "_top" || this.element != element.closest("turbo-frame") || id == "_top") {
  }
    return false
  }
``` 

It seemed odd to me to handle it one way for links and another for forms when the overall concept is very similar.  Almost seems like there's some shared code that should exist here.  

FYI, I've seen this happen a couple of times with turbo, where updates are made to link logic, but not forms or the other way around.

Reply:
I've opened https://github.com/hotwired/turbo/pull/402 and https://github.com/hotwired/turbo/pull/382 to attempt to reduce the surface area of the delegate interfaces, and to attempt to use consistent method naming across similar scenarios.

I don't think they'd address this particular issue entirely, but it's a start.

Reply:
I've got a piece of functionality ready to go that relies on the issue being fixed, so this PR getting merged would be helpful to me. Are those two PRs a prerequisite for this one to go through? I should also ask whether there's anything I can do to help progress this further.